dask/base.py

```diff
@@ -648,7 +649,10 @@ def tokenize(*args, **kwargs):
     """
     if kwargs:
         args = args + (kwargs,)
-    return md5(str(tuple(map(normalize_token, args))).encode()).hexdigest()
+    time_taken, result = get_wall_time(lambda: tuple(map(normalize_token, args)))
```

Comment:
We like to avoid lambdas if possible. You can move this `tuple(map(...` to a top-level function if needed.

Actually, it may be easier to just inline the time call checks here. We can refactor it later into its own method if needed.

Reply:
(small procedural note for the maintenance team: lambdas are particularly concerning in code that is placed into task graphs, because that code is more expensive to serialize than top level functions.  Lambdas are not such a big deal when talking about code that is run entirely locally)

Reply:
Good to know. `tokenize` is just done on the client?

Reply:
Yes, we commonly use it in collection code to generate keynames from arguments.  Once we have those keynames we're done with it.

It might also be called on a worker or something, but it's rare that we, from the client, will send tokenization code to the worker to do.  Even if we did that, it's ok as long as we're not passing a lambda itself.

```python
client.submit(lambda x: x + 1, ...) # bad

def f(x):
    g = lambda x: x + 1
    return g(x)

client.submit(f, ...)  # odd, but not a problem because the lambda isn't the thing that we're sending
```

Reply:
Thanks for the discussion here. I'm going to inline the calls.